### PR TITLE
[#552] Correct logic for choosing the CICD service

### LINKED ISSUE #552: Fix: Correct logic for choosing the CICD service
## Issue

Same issue with [choosing setup interface issue](https://github.com/nimblehq/ios-templates/commit/a5dd60bfccc913db6d6df6641290f5091453a733), when accidentally typing a different character, the process continues with any case contains that character instead of requesting to retry.
 
## Expected

- Only accept `g`, `b`, `c`, l, `github`, `bitrise`, `codemagic`, `later` with all upper and lower cases. Otherwise, requiring to type again.
 - Same with choosing workflow runner: `m`, `macOS`, `s`, `self-hosted`, `l`, `later`.
## Steps to reproduce

1.  Run script generating project.
2.  At `Which CI/CD service do you use (Can be edited later) [(g)ithub/(b)itrise/(c)odemagic/(l)ater]:` type a random character `r`

Screenshot 2023-12-29 at 16 37 51

### PR DESCRIPTION
**Pull request by nkhanh44**
- Close #552 

## What happened 👀

- Update the code in SetupCICD to fix the issue
- I added a description for choosing a runner of the GitHub service.

## Insight 📝

- N/a

## Proof Of Work 📹

Screenshot 2023-12-29 at 17 29 43

Screenshot 2023-12-29 at 16 55 15

### CODE REVIEW

File: Scripts/Swift/iOSTemplateMaker/Sources/iOSTemplateMaker/SetUpCICDService.swift [hunk 1]
```diff
@@ -7,34 +7,44 @@ struct SetUpCICDService {
...
+                "github": .github,
+                "b": .bitrise,
+                "bitrise": .bitrise,
+                "c": .codemagic,
+                "codemagic": .codemagic,
+                "l": .later,
+                "later": .later
+            ]
+
+            if let matchedCase = mappings[name] {
```
Comment (markgravity) [inline Scripts/Swift/iOSTemplateMaker/Sources/iOSTemplateMaker/SetUpCICDService.swift:22]: We can use `guard ` here 🙏

File: Scripts/Swift/iOSTemplateMaker/Sources/iOSTemplateMaker/SetUpCICDService.swift [hunk 2]
```diff
@@ -7,34 +7,44 @@ struct SetUpCICDService {
...
+                "m": .macOSLatest,
+                "macos": .macOSLatest,
+                "s": .selfHosted,
+                "self-hosted": .selfHosted,
+                "l": .later,
+                "later": .later
+            ]
+
+            let name = name.lowercased()
+            if let matchedCase = mappings[name] {
```
Comment (markgravity) [inline Scripts/Swift/iOSTemplateMaker/Sources/iOSTemplateMaker/SetUpCICDService.swift:45]: Same here 🙏

File: Scripts/Swift/iOSTemplateMaker/Sources/iOSTemplateMaker/SetUpCICDService.swift [hunk 3]
```diff
@@ -63,10 +73,12 @@ struct SetUpCICDService {
             fileManager.createDirectory(path: ".github/workflows")
             switch runnerType {
             case .macOSLatest:
+                print("Configured to run on the latest macOS.")
```
Comment (ducbm051291) [inline Scripts/Swift/iOSTemplateMaker/Sources/iOSTemplateMaker/SetUpCICDService.swift:76]: Please remove print

File: Scripts/Swift/iOSTemplateMaker/Sources/iOSTemplateMaker/SetUpCICDService.swift [hunk 4]
```diff
@@ -63,10 +73,12 @@ struct SetUpCICDService {
...
                 fileManager.moveFiles(in: ".github/project_workflows", to: ".github/workflows")
                 fileManager.removeItems(in: ".github/project_workflows")
                 fileManager.removeItems(in: ".github/self_hosted_project_workflows")
             case .selfHosted:
+                print("Configured to run on self-hosted.")
```
Comment (ducbm051291) [inline Scripts/Swift/iOSTemplateMaker/Sources/iOSTemplateMaker/SetUpCICDService.swift:81]: Please remove print

**Pull request closed without merging**